closingd/closingd.c

```diff
@@ -519,18 +527,35 @@ static struct amount_sat adjust_offer(struct per_peer_state *pps,
 			    type_to_string(tmpctx, struct amount_sat,
 					   &min_fee_to_accept));
 
-	/* Bisect between our minimum and max. */
-	if (amount_sat_greater(feerange->min, min_fee_to_accept))
-		min_fee_to_accept = feerange->min;
+	if (fee_negotiation_step_unit ==
+	    CLOSING_FEE_NEGOTIATION_STEP_UNIT_SATOSHI) {
+		/* -1 because the range boundary has already been adjusted with
+		 * one from our previous proposal. So, if the user requested a
+		 * step of 1 satoshi at a time we should just return our end of
+		 * the range from this function. */
+		step = (struct amount_sat){fee_negotiation_step - 1};
```

Comment:
`amount_sat_sub()` ? :)

Reply:
Do you mean to add a new function `amount_sat_from_u64()`? Are there other places in the code that would benefit from it?

Reply:
Ok I confused with `msat_from_u64()`... https://github.com/ElementsProject/lightning/blob/e99ab03f79bf71ab313c5b3e1d8446aa3fa45044/common/amount.c#L375

Reply:
That can still be used, but I think would be too cumbersome:

```c
struct amount_msat step_msat;
...
amount_msat_from_u64(&step_msat, (fee_negotiation_step - 1) * MSAT_PER_SAT);
step = amount_msat_to_sat_round_down(step_msat);
```

Reply:
Done that way with `amount_msat_from_u64()` because the typecast was upsetting `make check-amount-access`.

Reply:
I think you are still better off adding the helper for `sat`, but it can be done in a follow up cleanup, I think?